### ISSUE TITLE
Expose access to `vega.expressionFunction`

### ISSUE THREAD

**Issue by kanitw** [vega/vega-embed#934]
In https://github.com/vega/vega-embed/blob/next/src/embed.ts#L348, we already expose `formatLocale` and `timeFormatLocale` as `opts` for Vega embed. 

To let people specify [custom expression function](https://vega.github.io/vega/docs/api/extensibility/#expressionFunction), I think we should expose `expressionFunction` in our `opts` as well.

Since `vega.expressionFunction(name[, fn, visitor])` has a slightly more complicated signature than `formatLocale(definition) `, `timeFormatLocale(definition)`, we need to define `opt.expressionFunction`'s structure.

I think `{<name>: <fn>, <name2>: <fn2>}` as the short form and `{<name>: {fn: <fn>, visitor: <visitor>}, ...}` as the full form probably makes sense.

Basically, we'll add 
```
expressionFunction: ExpressionFunction
```

to `EmbedOptions`'s interface where the type is 

```ts
type ExpressionFunction = Record<string, Function | {fn: Function, visitor: Function}>;
```

@domoritz @jheer  -- any opinion? 

cc: @zacharybys

**Comment by domoritz** [comment]
Since this is pretty advanced, should we just provide the long form and not the shorthand? This makes the visitor more discoverable and reduces friction to add it. Yes, it's not as short for the commons case but I think that's okay for an advanced api. I'm happy either way if you think the shorthand is sooooo much more common.

**Comment by kanitw** [comment]
FWIW, I have only used the fn, but never have to use the visitor so far, I guess that most people will only need the shorthand.

**Comment by kanitw** [comment]
I think another argument for the shorthand is that it can provide consistency if we later add `projection(s): {<name>: <projection>, <name2>: <projection2>}`.

**Comment by domoritz** [comment]
Okay. I'm convinced to have the shorthand. The design makes sense to me.

**Comment by kanitw** [comment]
For the prop name, should we do `expressionFunction: {...}` or `expressionFunctions: {...}` or sth else ?

Basically the function is `vega.expressionFunction`, but the object passed in technically can represent multiple expressionFunction.

There is also `expr:`  for "Custom Vega Expression interpreter" (which is basically the same as `expr:`  in `vega.View`).

I think I prefer `expressionFunction:` so that the name exactly matches the function so it's easier to know that it's a wrapper for the same thing.

**Comment by kanitw** [comment]
fixed in #935
